### PR TITLE
Sourcery Starbot ⭐ refactored liaoyuhua/GPT-TS

### PR DESCRIPTION
**Pull request by SourceryAI**
Thanks for starring [sourcery-ai/sourcery](/sourcery-ai/sourcery) ✨ 🌟 ✨

Here's your pull request refactoring your most popular Python repo.

If you want Sourcery to refactor all your Python repos and incoming pull requests [install our bot](/apps/sourcery-ai/installations/new).

<details>
<summary>Review changes via command line</summary>
<p>To manually merge these changes, make sure you're on the <code>main</code> branch, then run:</p>
<pre>
git fetch https://github.com/sourcery-ai-bot/GPT-TS main
git merge --ff-only FETCH_HEAD
git reset HEAD^
</pre>
</details>

### CODE REVIEW

File: src/dataset.py [hunk 1]
```diff
@@ -24,7 +24,7 @@ def __init__(
     ):
         # size [seq_len, pred_len]
         # info
-        if size == None:
+        if size is None:
```
Comment (SourceryAI) [inline src/dataset.py:27]: Function `Dataset_ETT_hour.__init__` refactored with the following changes:

- Use x is None rather than x == None ([`none-compare`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/none-compare/))

File: src/dataset.py [hunk 2]
```diff
@@ -63,7 +63,7 @@ def __read_data__(self):
         border1 = border1s[self.set_type]
         border2 = border2s[self.set_type]
 
-        if self.features == "M" or self.features == "MS":
+        if self.features in ["M", "MS"]:
```
Comment (SourceryAI) [inline src/dataset.py:66]: Function `Dataset_ETT_hour.__read_data__` refactored with the following changes:

- Replace multiple comparisons of same variable with `in` operator ([`merge-comparisons`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/merge-comparisons/))

File: src/dataset.py [hunk 3]
```diff
@@ -127,7 +127,7 @@ def __init__(
     ):
         # size [seq_len, pred_len]
         # info
-        if size == None:
+        if size is None:
```
Comment (SourceryAI) [inline src/dataset.py:130]: Function `Dataset_ETT_minute.__init__` refactored with the following changes:

- Use x is None rather than x == None ([`none-compare`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/none-compare/))

File: src/dataset.py [hunk 4]
```diff
@@ -166,7 +166,7 @@ def __read_data__(self):
         border1 = border1s[self.set_type]
         border2 = border2s[self.set_type]
 
-        if self.features == "M" or self.features == "MS":
+        if self.features in ["M", "MS"]:
```
Comment (SourceryAI) [inline src/dataset.py:169]: Function `Dataset_ETT_minute.__read_data__` refactored with the following changes:

- Replace multiple comparisons of same variable with `in` operator ([`merge-comparisons`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/merge-comparisons/))

File: src/loss.py [hunk 5]
```diff
@@ -11,8 +11,7 @@ def mae_loss(prediction, target):
     mask = ~torch.isnan(target)
     masked_prediction = prediction[mask]
     masked_target = target[mask]
-    loss = F.l1_loss(masked_prediction, masked_target)
-    return loss
+    return F.l1_loss(masked_prediction, masked_target)
```
Comment (SourceryAI) [inline src/loss.py:14]: Function `mae_loss` refactored with the following changes:

- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/inline-immediately-returned-variable/))

File: src/loss.py [hunk 6]
```diff
@@ -24,5 +23,4 @@ def mse_loss(prediction, target):
     mask = ~torch.isnan(target)
     masked_prediction = prediction[mask]
     masked_target = target[mask]
-    loss = F.mse_loss(masked_prediction, masked_target)
-    return loss
+    return F.mse_loss(masked_prediction, masked_target)
```
Comment (SourceryAI) [inline src/loss.py:26]: Function `mse_loss` refactored with the following changes:

- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/inline-immediately-returned-variable/))

File: src/model.py [hunk 7]
```diff
@@ -192,7 +192,7 @@ def from_pretrained(cls, config: dict):
             f"h.{i}." for i in range(n_layer, model_hf.config.n_layer)
         ]
         sd_hf = {
-            k: v for k, v in sd_hf.items() if not any((k_ in k) for k_ in drop_keys)
+            k: v for k, v in sd_hf.items() if all(k_ not in k for k_ in drop_keys)
```
Comment (SourceryAI) [inline src/model.py:195]: Function `GPT2TS.from_pretrained` refactored with the following changes:

- Invert any/all to simplify comparisons ([`invert-any-all`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/invert-any-all/))

File: src/modules.py [hunk 8]
```diff
@@ -39,11 +39,11 @@ def __init__(self, num_features: int, eps=1e-5, affine=True):
...
 
     def forward(self, x: torch.Tensor, mode: str):
-        if mode == "norm":
+        if mode == "denorm":
+            x = self._denormalize(x)
+        elif mode == "norm":
             self._get_statistics(x)
             x = self._normalize(x)
-        elif mode == "denorm":
-            x = self._denormalize(x)
```
Comment (SourceryAI) [inline src/modules.py:46]: Function `RevIN.forward` refactored with the following changes:

- Simplify conditional into switch-like form ([`switch`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/switch/))

File: src/modules.py [hunk 9]
```diff
@@ -76,8 +76,7 @@ def _denormalize(self, x: torch.Tensor):
             x = x - self.affine_bias
             x = x / (self.affine_weight + self.eps * self.eps)
         x = x * self.stdev
-        x = x + self.mean
-        return x
+        return x + self.mean
```
Comment (SourceryAI) [inline src/modules.py:79]: Function `RevIN._denormalize` refactored with the following changes:

- Inline variable that is immediately returned ([`inline-immediately-returned-variable`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/inline-immediately-returned-variable/))

File: src/modules.py [hunk 10]
```diff
@@ -231,7 +230,7 @@ def __init__(self, individual, n_vars, nf, target_window, head_dropout=0.1):
             self.linears = nn.ModuleList()
             self.dropouts = nn.ModuleList()
             self.flattens = nn.ModuleList()
-            for i in range(self.n_vars):
+            for _ in range(self.n_vars):
```
Comment (SourceryAI) [inline src/modules.py:233]: Function `FlattenHead.__init__` refactored with the following changes:

- Replace unused for index with underscore ([`for-index-underscore`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/for-index-underscore/))

File: src/utils.py [hunk 11]
```diff
@@ -33,7 +33,7 @@ def save_checkpoint(self, val_loss, model, path):
             print(
                 f"Validation loss decreased ({self.val_loss_min:.6f} --> {val_loss:.6f}).  Saving model ..."
             )
-        torch.save(model.state_dict(), path + "/" + "checkpoint.pth")
+        torch.save(model.state_dict(), f"{path}/checkpoint.pth")
```
Comment (SourceryAI) [inline src/utils.py:36]: Function `EarlyStopping.save_checkpoint` refactored with the following changes:

- Use f-string instead of string concatenation [×2] ([`use-fstring-for-concatenation`](https://docs.sourcery.ai/Reference/Default-Rules/refactorings/use-fstring-for-concatenation/))

**Comment by liaoyuhua** [comment]
Optimized the code formatting.
